src/main/java/rest/StudentRessource.java

```diff
@@ -62,26 +62,34 @@ public String data() {
     public String demo() {
         return "{\"msg\":\"path students succesful\"}";
     }
-    
+
     @GET
     @Path("/allstudents")
     @Produces({MediaType.APPLICATION_JSON})
     public String getAllStudents() throws Exception {
         try {
-            return GSON.toJson(FACADE.getAllStudentDTO());
+            if (FACADE.getAllStudentDTO() == null || FACADE.getAllStudentDTO().isEmpty()) {
+                return "{\"error\":\"no students exists in the database\"}";
+            } else {
+                return GSON.toJson(FACADE.getAllStudentDTO());
+            }
```

Comment:
I think you should call .getAllStudentDTO() once and check the result afterwards and then return something based on the result (Like Rúni suggested earlier). Right now the .getAllStudentDTO() method is being called 3 times in a successful case.